src/server/api/routes.rs

```diff
@@ -0,0 +1,275 @@
+    state: &T,
+) -> anyhow::Result<
+    App<
+        impl ServiceFactory<
+            ServiceRequest,
+            Response = ServiceResponse<U>,
+            Config = (),
+            InitError = (),
+            Error = Error,
+        >,
+    >,
+> {
+    app = app
+        .service(
+            web::scope("/_api").service(
+                web::scope("/versions")
+                    .service(
+                        web::resource(
+                            "/_publication/{publication}/_compare/{date}/{compare_date}/{path:.*}",
+                        )
+                        .to(versions),
+                    )
+                    .service(web::resource("/_publication/{publication}/_date/{date}").to(versions))
+                    .service(web::resource("/_publication/{publication}").to(versions))
+                    .service(web::resource("/_publication/{publication}/{path:.*}").to(versions))
+                    .service(
+                        web::resource("/_compare/{date}/{compare_date}/{path:.*}").to(versions),
+                    )
+                    .service(web::resource("/_date/{date}/{path:.*}").to(versions))
+                    .service(web::resource("/{path:.*}").to(versions)),
+            ),
+        )
+        .app_data(web::Data::new(state.clone()));
+
+    let initialized_app = register_dynamic_routes(app, state)?;
+    Ok(initialized_app)
+}
+
+/// Initialize all dynamic routes for the given Archive.
+///
+/// Dynamic routes are determined at runtime by looking at the stele's `dependencies.json` and `repositories.json` files
+/// in the authentication (e.g. law) repository.
+///
+/// # Errors
+/// Errors if unable to register dynamic routes (e.g. if git repository cannot be opened)
+fn register_dynamic_routes<T: MessageBody>(
+    mut app: App<
+        impl ServiceFactory<
+            ServiceRequest,
+            Response = ServiceResponse<T>,
+            Config = (),
+            InitError = (),
+            Error = Error,
+        >,
+    >,
+    state: &impl Global,
+) -> anyhow::Result<
+    App<
+        impl ServiceFactory<
+            ServiceRequest,
+            Response = ServiceResponse<T>,
+            Config = (),
+            InitError = (),
+            Error = Error,
+        >,
+    >,
+> {
+    let config = state.archive().get_config()?;
+    let stelae_guard = config
+        .headers
+        .and_then(|headers| headers.current_documents_guard);
+
+    if let Some(guard) = stelae_guard {
+        tracing::info!(
+            "Initializing guarded current documents with header: {}",
+            guard
+        );
+        HEADER_NAME.get_or_init(|| guard);
+        HEADER_VALUES.get_or_init(|| {
+            state
+                .archive()
+                .stelae
+                .keys()
+                .map(ToString::to_string)
+                .collect()
+        });
+
+        if let (Some(guard_name), Some(guard_values)) = (HEADER_NAME.get(), HEADER_VALUES.get()) {
```

Comment:
Then this can be a function guard clause:

```rust
let (Some(guard_name), Some(guard_values)) = (HEADER_NAME.get(), HEADER_VALUES.get())
else {
    // logging ...
    return ...;
};
```

Reply:
Wouldn't compile with `let` instead of `if let`, but otherwise added logging and return statement

Reply:
It wouldn't compile? That's not because you've got an old version of Rust? Here's the syntax I was referring to https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html